core/ecschema-metadata/src/Deserialization/XmlParser.ts

```diff
@@ -1032,7 +1032,8 @@ export class XmlParser extends AbstractParser<Element> {
 
   private addCAPropertyValues(caClass: CustomAttributeClass, propertyElements: Element[]): CustomAttribute {
     const instance = { className: caClass.fullName } as CustomAttribute;
-    if (!caClass.properties)
+    // TODO: is this a bug? Shouldn't we be checking for all properties on the class?
+    if (caClass.getPropertiesSync(true)[Symbol.iterator]().next().done)
```

Comment:
do we even need this?  I guess it would improve performance IF the CA has no properties but the serialized object does.  Unless you see some problem I suggest we just remove this check

Reply:
Agreed, I removed it